connectors/stash/src/main/java/com/vmware/connectors/stash/StashController.java

```diff
@@ -0,0 +1,444 @@
+        consumes = MediaType.APPLICATION_FORM_URLENCODED_VALUE
+    )
+    public Single<ResponseEntity<String>> approve(final StashPullRequest pullRequest,
+                                                  @RequestHeader(AUTH_HEADER) final String authHeader,
+                                                  @RequestHeader(BASE_URL_HEADER) final String baseUrl) {
+
+        logger.debug("Approve action for stash pull request: {}, baseURL: {}",
+                pullRequest,
+                baseUrl);
+
+        return performStashAction(baseUrl, authHeader, pullRequest, StashAction.APPROVE);
+    }
+
+    @PostMapping(
+            path = "/api/v1/{projectKey}/{repositorySlug}/{pullRequestId}/merge",
+            consumes = MediaType.APPLICATION_FORM_URLENCODED_VALUE
+    )
+    public Single<ResponseEntity<String>> merge(final StashPullRequest pullRequest,
+                                                  @RequestHeader(AUTH_HEADER) final String authHeader,
+                                                  @RequestHeader(BASE_URL_HEADER) final String baseUrl) {
+
+        logger.debug("Merge action for stash pull request: {}, baseURL: {}",
+                pullRequest,
+                baseUrl);
+
+        return performStashAction(baseUrl, authHeader, pullRequest, StashAction.MERGE);
+    }
+
+    @PostMapping(
+            path = "/api/v1/{projectKey}/{repositorySlug}/{pullRequestId}/decline",
+            consumes = MediaType.APPLICATION_FORM_URLENCODED_VALUE
+    )
+    public Single<ResponseEntity<String>> decline(final StashPullRequest pullRequest,
+                                                @RequestHeader(AUTH_HEADER) final String authHeader,
+                                                @RequestHeader(BASE_URL_HEADER) final String baseUrl) {
+
+        logger.debug("Decline action for stash pull request: {}, baseURL: {}",
+                pullRequest,
+                baseUrl);
+
+        return performStashAction(baseUrl, authHeader, pullRequest, StashAction.DECLINE);
+    }
+
+    @PostMapping(
+            path = "/api/v1/{projectKey}/{repositorySlug}/{pullRequestId}/comments",
+            consumes = MediaType.APPLICATION_FORM_URLENCODED_VALUE
+    )
+    public Single<ResponseEntity<String>> comments(final StashPullRequest pullRequest,
+                                                  @RequestParam(COMMENT_PARAM_KEY) final String comment,
+                                                  @RequestHeader(AUTH_HEADER) final String authHeader,
+                                                  @RequestHeader(BASE_URL_HEADER) final String baseUrl) {
+
+        logger.debug("Comment action for stash pull request: {}, baseURL: {}",
+                pullRequest,
+                baseUrl);
+
+        final HttpHeaders headers = buildHeaders(authHeader);
+        headers.add(HttpHeaders.CONTENT_TYPE, MediaType.APPLICATION_JSON_VALUE);
+
+        final String url = String.format(StashConstants.STASH_PR_COMMENT_URL_FORMAT,
```

Comment:
It would be better to use spring's UriComponentsBuilder instead of String.format.  Spring will escape/encode the url for us (reducing risks of malicious input causing us to call out to something unexpected).

Reply:
I have used ```UriComponentsBuilder``` instead of ```String.format``` in all the places..